whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapObjectMapper.java

```diff
@@ -332,7 +329,24 @@ private static Autnum createAutnumResponse(final RpslObject rpslObject) {
         final Autnum autnum = new Autnum();
         autnum.setHandle(rpslObject.getKey().toString());
         autnum.setName(rpslObject.getValueForAttribute(AttributeType.AS_NAME).toString().replace(" ", ""));
-        autnum.setType("DIRECT ALLOCATION");
+        autnum.setType(DIRECT_ALLOCATION);
+        autnum.getEntitySearchResults().addAll(createContactEntities(rpslObject));
+        return autnum;
+    }
+
+    private static Autnum createAsBlockResponse(final RpslObject rpslObject) {
+        final Autnum autnum = new Autnum();
+
+        final String key = rpslObject.getValueForAttribute(AttributeType.AS_BLOCK).toString();
+        final String handle = key.split("-")[0].replace(" ", "");
```

Comment:
Can you read the handle from the blockRange below? i.e. blockRange.getBegin()

Reply:
Handle property has prefix AS. AsRange only holds numeric value

Reply:
There's no need to also parse the key, use the AsRange and convert to a String. It's less logic and easier to understand (imo).
```
final String handle = Long.toString(blockRange.getBegin());
```